lib/Twig/Error.php

```diff
@@ -67,8 +69,9 @@ public function __construct($message, $lineno = -1, $name = null, Exception $pre
 
         $this->lineno = $lineno;
         $this->filename = $name;
+        $this->path = $path;
 
-        if (-1 === $lineno || null === $name) {
+        if (-1 === $lineno || null === $name || null === $path) {
```

Comment:
also note that the default return value of getPath is the empty string, not null.